Mathlib/GroupTheory/MonoidLocalization.lean

```diff
@@ -1886,6 +1901,66 @@ noncomputable def lift (f : LocalizationWithZeroMap S N) (g : M →*₀ P)
       exact f.toMonoidWithZeroHom.map_zero.symm }
 #align submonoid.localization_with_zero_map.lift Submonoid.LocalizationWithZeroMap.lift
 
+/-- Given a Localization map `f : M →*₀ N` for a Submonoid `S ⊆ M`,
+if `M` is left cancellative monoid with zero, and all elements of `S` are
+left regular, then N is a left cancellative monoid with zero. -/
+theorem leftCancelMulZero_of_le_isLeftRegular
+    (f : LocalizationWithZeroMap S N) [IsLeftCancelMulZero M]
+    (h : ∀ ⦃x⦄, x ∈ S → IsLeftRegular x) : IsLeftCancelMulZero N := by
+  let fl := f.toLocalizationMap
+  let g := f.toMap
+  --- We prove the property to be left cancellative.
+  have mul_cancel (a z w : N): a ≠ 0 → a * z = a * w → z = w := by
+      intro ha hazw
+      --- Using `LocalizationMap.surj` we find an equality in M that
+      --- imply the goal.
+      cases' LocalizationMap.surj fl a with b hb
```

Comment:
Can you use `cases` instead of `cases'` ?

Reply:
Yes, I can. But, pardon my ignorance, ¿is it preferable?

Reply:
I could use also  `obtain ⟨b, hb ⟩ := LocalizationMap.surj fl a `

Reply:
Yes, `cases'` (and all the various `tac'`) is from Lean3, but sooner or later it will disappear. See for example #9171 .

Reply:
I changed, but with obtain, which I like more...